### PR TITLE
Hive: Add support for EXCHANGE PARTITION in ALTER TABLE

### PR DESCRIPTION
**Pull request by nahuelverdugo**
### Brief summary of the change made
Added support for parsing `Hive` queries when a partition is exchanged between tables in an `ALTER TABLE` statement. More info in the Hive [doc](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-ExchangePartition).

Some minor changes in the contribution docs after following it.

### Are there any other side effects of this change that we should be aware of?
I don't think so.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_hive.py [hunk 1]
```diff
@@ -1017,3 +1017,55 @@ class SortByClauseSegment(ansi.OrderByClauseSegment):
...
+                        Ref("BracketedColumnReferenceListGrammar"),
+                        optional=True,
+                    ),
+                ),
+                # Rename
+                Sequence(
+                    "RENAME",
+                    OneOf("AS", "TO", optional=True),
+                    Ref("TableReferenceSegment"),
+                ),
```
Comment (greg-finley) [inline src/sqlfluff/dialects/dialect_hive.py:1060]: This part seems all the same as ANSI. I wonder if we could redefine that as `AlterTableOptionsGrammar` or something in ANSI and then just extend in Hive to add the "exchange" part, similar to: https://github.com/sqlfluff/sqlfluff/blob/72fed8048248426a7b6e95880cec87641249db06/src/sqlfluff/dialects/dialect_hive.py#L157-L166

**Review by greg-finley — COMMENTED** [review]
Nice start! But I feel we can reduce code duplication a bit.

File: src/sqlfluff/dialects/dialect_hive.py (hunk 1 above)
Comment (nahuelverdugo) [reply src/sqlfluff/dialects/dialect_hive.py:1060]: Hi @greg-finley thank you for the suggestion, should I do the same replacement in other dialects where the `ansi` `AlterTableStatementSegment` class is overwritten?

**Review by greg-finley — APPROVED** [review]
Thanks for your contribution, and for addressing the feedback! 🙌

File: src/sqlfluff/dialects/dialect_hive.py (hunk 1 above)
Comment (greg-finley) [reply src/sqlfluff/dialects/dialect_hive.py]: If you find similar ways that they are copying the same code, sure! It can be a separate PR.

**Pull request merged**